src/common_swagger_api/schema/quicklaunches.clj

```diff
@@ -0,0 +1,104 @@
+(ns common-swagger-api.schema.quicklaunches
+  (:use [common-swagger-api.schema :only [describe NonBlankString]])
+  (:require [schema.core :as s]
+            [schema-tools.core :as st]
+            [common-swagger-api.schema.apps :refer [AnalysisSubmission]])
+  (:import [java.util UUID]))
+
+(s/defschema Submission
+  {:id
+   (describe UUID "The UUID for this submission")
+
+   :submission
+   AnalysisSubmission})
+
+(s/defschema NewSubmission
+  (st/dissoc Submission :id))
+
+(s/defschema QuickLaunch
+  {:id
+   (describe UUID "The UUID for the quick launch")
+
+   :name
+   (describe NonBlankString "The name of the quick launch")
+
+   (s/optional-key :description)
+   (describe String "The description of the quick launch")
+
+   :creator
+   (describe NonBlankString "The username for the creator of the quick launch")
+
+   :app_id
+   (describe UUID "The UUID for the app the quick launch is associated with")
+
+   (s/optional-key :is_public)
+   (describe Boolean "Whether the quick launch is publicly available. Defaults to false")
+
+   :submission
+   AnalysisSubmission})
+
+(s/defschema NewQuickLaunch
+  (st/dissoc QuickLaunch :id :user)) ;user should be included in the request query params
+
+(s/defschema UpdateQuickLaunch
+  (-> QuickLaunch
+    (st/dissoc :id)
+    (st/optional-keys-schema)))
+
+(s/defschema QuickLaunchFavorite
+  {:id
+   (describe UUID "The UUID for the quick launch favorite entry")
+
+   :quick_launch_id
+   (describe UUID "The UUID of the quick launch that was favorited")
+
+   :user
+   (describe NonBlankString "The username of the user that favorited the quick launch")})
+
+(s/defschema NewQuickLaunchFavorite
+  (st/dissoc QuickLaunchFavorite :id :user)) ;user is in query params, id is auto-assigned.
+
+; Mostly just useful for admin operations. Users will either create or remove favorites.
+(s/defschema UpdateQuickLaunchFavorite
```

Comment:
I'd say save ourselves some work and don't even bother with an update endpoint for favorites, until we really need one.